File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        unordered_map<int,int>mpp;
        set<int>st(arr.begin(),arr.end());
        int rank=1;
        for(auto x:st) mpp[x]=rank++;
        vector<int>ans;
        for(int &x:arr) ans.emplace_back(mpp[x]);
        return ans;
    }
};

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cs
public class Solution {
    public int[] ArrayRankTransform(int[] arr) {
        int n=arr.Length;
        int[] ans=new int[n];
        SortedSet<int> st=new SortedSet<int>();
        foreach(int x in arr) st.Add(x);
        int rank=1;
        Dictionary<int,int>mpp=new Dictionary<int,int>();
        foreach(var x in st) mpp[x]=rank++;
        for(int i=0;i<n;i++){
            ans[i]=mpp[arr[i]];
        }
        return ans;
    }
}